crates/sui-framework/packages/sui-framework/sources/accumulator.move

```diff
@@ -17,3 +21,110 @@ fun create(ctx: &TxContext) {
         id: object::sui_accumulator_root_object_id(),
     })
 }
+
+/// Balance object methods
+
+/// Key is used only for computing the field id of accumulator objects.
+/// `T` is the type of the accumulated value, e.g. Balance<SUI>
+public struct Key<phantom T> has copy, drop, store {
+    address: address,
+}
+
+fun root_has_accumulator<K, V: store>(accumulator_root: &AccumulatorRoot, name: Key<K>): bool {
+    dynamic_field::exists_with_type<Key<K>, V>(&accumulator_root.id, name)
+}
+
+use fun root_has_accumulator as AccumulatorRoot.has_accumulator;
+
+fun root_add_accumulator<K, V: store>(
+    accumulator_root: &mut AccumulatorRoot,
+    name: Key<K>,
+    value: V,
+) {
+    dynamic_field::add(&mut accumulator_root.id, name, value);
+}
+
+use fun root_add_accumulator as AccumulatorRoot.add_accumulator;
+
+fun root_borrow_accumulator_mut<K, V: store>(
+    accumulator_root: &mut AccumulatorRoot,
+    name: Key<K>,
+): &mut V {
+    dynamic_field::borrow_mut<Key<K>, V>(&mut accumulator_root.id, name)
+}
+
+use fun root_borrow_accumulator_mut as AccumulatorRoot.borrow_accumulator_mut;
+
+fun root_remove_accumulator<K, V: store>(accumulator_root: &mut AccumulatorRoot, name: Key<K>): V {
```

Comment:
Could you categorize these functions for which you see as being 
- temporary for testing
- used in settlement
- intended to be used by "accumulators"